### DISCUSSION TITLE
Automatically pull if there is a newer base image?

### DISCUSSION THREAD

**Discussion by GrabbenD** [containers/podman#20121] [Q&A] [answered]
Is it possible to automatically detect updates for upstream images when doing `podman build` of a local `Containerfile`?

**Example:**
1. `$ less Containerfile`
   ```Dockerfile
   FROM archlinux:base
   RUN echo OK
   ```
2. `$ podman build -f Containerfile`
3. Skip cache if `archlinux:base` image is updated upstream

**Comment by vrothberg** [comment, marked as answer]
Thanks for reaching out.  Have a look at the [--pull flag](https://docs.podman.io/en/latest/markdown/podman-build.1.html#pull-policy).

**Reply by GrabbenD** [reply]
Thank you!

I find this highly confusing. It says the default is `always` for `--pull`. However, if I don't specify `--pull=always` it just uses the cache (it completes instantly) but if I do specify `--pull=always` it pauses for a second and actually checks if that's the newest (it even says `Copying blob 38f5a258dd4f skipped: already exists`).

If `always` is the default policy, why do I have to specify `--pull=always` for it to work?

**Reply by vrothberg** [reply]
Thanks for pointing that out, @GrabbenD.

I can confirm that the default is "missing".  The man pages need to be updated.  I will open a PR to fix that in a moment.

**Reply by GrabbenD** [reply]
That explains it, thanks!

Does `newer` option actually do anything?
There's no confirmation that current digests matches upstream in the logs (for reference: I'm using using `:latest`) 😕

**Reply by vrothberg** [reply]
`newer` only pulls the image if the upstream digest is different than the one in the local storage.

**Reply by GrabbenD** [reply]
Gotcha!

While playing around with this I noticed that `-t name` automatically tags as `localhost/name:latest`, if you specify `FROM localhost/name:latest` or `FROM name:latest` in `Containerfile` and use `--pull=always` it fails due to `WARN[0000] Failed, retrying in 2s ... (1/3). Error: initializing source docker://localhost/name:latest: pinging container registry localhost: Get "https://localhost/v2/": dial tcp [::1]:443: connect: connection refused` which isn't the case if you never specify `--pull` or use `--pull=newer`

**Reply by GrabbenD** [reply]
I went ahead and made bug reports for visibility
- https://github.com/containers/podman/issues/20125
- https://github.com/containers/podman/issues/20126
